packages/@ember/debug/lib/deprecate.ts

```diff
@@ -187,6 +192,28 @@ if (DEBUG) {
     assert(missingOptionsIdDeprecation, Boolean(options!.id));
     assert(missingOptionsUntilDeprecation, Boolean(options!.until));
 
+    if (!options.for) {
+      deprecate(missingOptionsForDeprecation, Boolean(options!.for), {
+        id: 'ember-source.deprecation-without-for',
+        until: '4.0.0',
+        for: 'ember-source',
+        since: {
+          available: '3.22.0',
+        },
+      });
+    }
+
+    if (!options.since) {
+      deprecate(missingOptionsSinceDeprecation, Boolean(options!.since), {
+        id: 'ember-source.deprecation-without-since',
+        until: '4.0.0',
+        for: 'ember-source',
+        since: {
+          available: '3.22.0',
+        },
+      });
+    }
```

Comment:
Same here, RE: the two above points:

* use a debug only `Set` to avoid issuing the same deprecation many many times
* include information about the original deprecation id that is triggering the "knock on" deprecation